### PR TITLE
Require 3 active config servers/controllers when allowing removal

### PR DESCRIPTION
**Pull request by hakonhall**
Makes the following changes to the completion of retirement of a config server or controller node:
 - require all 3 nodes to be active
 - disallow automatic expiry after 4 days

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java [hunk 1]
```diff
@@ -93,7 +95,32 @@ private boolean canRemove(Node node) {
...
+
+            if (activeNodes.nodeType(node.type()).asSet().size() < NUM_CONFIG_SERVERS) {
+                // Scenario:  All 3 config servers want to retire.
+                //
+                // Say RetiredExpirer runs on cfg1 and gives cfg2 permission to be removed (PERMANENTLY_DOWN in ZK).
+                // The consequent redeployment moves cfg2 to inactive, removing cfg2 from the application,
+                // and PERMANENTLY_DOWN for cfg2 is cleaned up.
+                //
+                // If the RetiredExpirer on cfg3 now runs before its InfrastructureProvisioner, then
+                //  a. But the duper model still contains cfg2
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java:109]: s/But //

**Review by mpolden — CHANGES_REQUESTED** [review]
Good, but this needs a unit test.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java (hunk 1 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java]: Done

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java [hunk 2]
```diff
@@ -93,7 +95,32 @@ private boolean canRemove(Node node) {
             return false;
         }
 
-        if (node.history().hasEventBefore(History.Event.Type.retired, clock().instant().minus(retiredExpiry))) {
+        if (node.type().isConfigServerLike()) {
+            // Avoid eventual expiry of configserver-like nodes
+
+            if (activeNodes.nodeType(node.type()).asSet().size() < NUM_CONFIG_SERVERS) {
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java:101]: Nit: `NodeList` has `size` so need to copy into a set.

**Review by mpolden — APPROVED** [review]
Great!

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java (hunk 2 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java]: Fixed

**Review by mpolden — APPROVED** [review]

**Pull request merged**